public/catch-js-errors.js

```diff
@@ -0,0 +1,17 @@
+function sendErrorsToAPI( message, scriptUrl, lineNumber, columnNumber, error ) {
+	var xhr = new XMLHttpRequest();
+
+	error = error || new Error( message );
+
+	// POST to the API
+	xhr.open( 'POST', 'https://public-api.wordpress.com/js-error', true );
+	xhr.setRequestHeader( 'Content-type', 'application/x-www-form-urlencoded' );
+	xhr.send( 'message=' + error.message + '&type=' + error.constructor && error.constructor.name + '&userAgent=' + navigator.userAgent + '&stack=' + error.stack );
```

Comment:
would it not also make some sense to just send this all as structured data in the POST body? something like a JSON export or just the plain message from the exception?